### ISSUE TITLE
Choose between IPv4 and IPv6

### ISSUE THREAD

**Issue by boptico** [ivpn/desktop-app#393]
Hello IVPN team,
Please let us choose between IPv4 and IPv6 when enabled, so that if one IP is blocked, the other can be used.
Thanks.

**Comment by stenya** [comment]
Hi,
The configuration parameter "Enable IPv6 in VPN tunnel" applies only to communication through the VPN tunnel.
To use it, your client connects to the VPN server using the IPv4 protocol, after which IPv6 is configured inside the tunnel.

This means that IPv4 connectivity will still be used and must not be blocked.

**Comment by boptico** [comment]
@stenya I was specifically asking about the IP address that websites see, not about disabling IPv4/6 completely. Is this possible?

**Comment by stenya** [comment]
I assume you are asking whether another protocol (IPv4/IPv6) will be used if one of the user's IP addresses is blocked on the web server's side.

When a VPN connection is established (with IPv6 enabled), both protocols (IPv4 and IPv6) are in use. The decision on which protocol to use for a particular connection depends on various factors and is determined by your environment (OS, web browser).

In general, if both the client and the web server support both protocols, and one of the protocols (or the client's IP) is blocked, it is likely that the other (unblocked) protocol will be used.

**Comment by stenya** [comment]
Please clarify:
You want to enable an "IPv6-only" mode inside the VPN tunnel, where IPv4 is disabled. Is that correct?

---
_Currently, it is possible to use "IPv4-only" and "IPv4+IPv6" modes.
For IPv4+IPv6, the decision on which protocol to use for a particular connection depends on various factors and is determined by your environment (OS, web browser).
In general, if both the client and the web server support both protocols and one of the protocols (or the client's IP) is blocked, it is likely that the other (unblocked) protocol will be used._

**Comment by boptico** [comment]
Yes, I mean that all VPN traffic would only use IPv6. So if I go to any website, it would see my IPv6 address, always.

**Comment by ng8weji9vowdv** [comment]
@stenya according to the faq you on the websites you guys switch ip addresses sometimes, but its been almost a couple weeks of me not being able to access github on most of the ivpn locations, and been months of me not being able to access icloud login form either idmsa.apple.com, and web archive org neither (many more including yahoo but doesnt matter), do you guys actually pull new ip addresses or not? being able to switch to ipv6 couldve resolved this issue.

**Comment by ng8weji9vowdv** [comment]
@quite as well.

**Comment by stenya** [comment]
> Yes, I mean that all VPN traffic would only use IPv6. So if I go to any website, it would see my IPv6 address, always.

For now, this functionality is not a priority, but we will consider it in the future.

**Comment by ng8weji9vowdv** [comment]
<details open>
<summary>Results</summary>

```
br.gw.ivpn.net [223ms | OpenVPN] github.com s (ping) | web.archive.org f (ping) | idmsa.apple.com f (ping) | idmsa.apple.com s (ec: 8) | web.archive.org s |
ro.gw.ivpn.net [67ms | OpenVPN] github.com s (ping) | web.archive.org f (ping) | idmsa.apple.com f (ping) | idmsa.apple.com s (ec: 8) | web.archive.org s |
us-co.gw.ivpn.net [157ms | OpenVPN] github.com s (ping) | web.archive.org f (ping) | idmsa.apple.com f (ping) | idmsa.apple.com s (ec: 8) | web.archive.org s |
is.gw.ivpn.net [79ms | OpenVPN] github.com s (ping) | web.archive.org f (ping) | idmsa.apple.com s (ping) | idmsa.apple.com s (ec: 8) | web.archive.org s |
us-tx.gw.ivpn.net [152ms | OpenVPN] github.com s (ping) | web.archive.org f (ping) | idmsa.apple.com s (ping) | idmsa.apple.com s (ec: 8) | web.archive.org s |
us-ut.gw.ivpn.net [191ms | OpenVPN] github.com s (ping) | web.archive.org f (ping) | idmsa.apple.com s (ping) | idmsa.apple.com s (ec: 8) | web.archive.org s |
hr.gw.ivpn.net [70ms | OpenVPN] github.com s (ping) | web.archive.org f (ping) | idmsa.apple.com f (ping) | idmsa.apple.com s (ec: 8) | web.archive.org s |
pt.wg.ivpn.net [84ms | WireGuard] github.com s (ping) | web.archive.org f (ping) | idmsa.apple.com f (ping) | idmsa.apple.com s (ec: 4) | web.archive.org s |
ca-bc.gw.ivpn.net [166ms | OpenVPN] github.com s (ping) | web.archive.org f (ping) | idmsa.apple.com f (ping) | idmsa.apple.com s (ec: 8) | web.archive.org s |
pe.wg.ivpn.net [207ms | WireGuard] github.com s (ping) | web.archive.org f (ping) | idmsa.apple.com f (ping) | idmsa.apple.com s (ec: 8) | web.archive.org s |
pl.gw.ivpn.net [75ms | OpenVPN] github.com s (ping) | web.archive.org f (ping) | idmsa.apple.com s (ping) | idmsa.apple.com s (ec: 8) | web.archive.org s |
ch.wg.ivpn.net [51ms | WireGuard] github.com s (ping) | web.archive.org f (ping) | idmsa.apple.com f (ping) | idmsa.apple.com s (ec: 8) | web.archive.org s |
hu.gw.ivpn.net [84ms | OpenVPN] github.com s (ping) | web.archive.org f (ping) | idmsa.apple.com s (ping) | idmsa.apple.com s (ec: 8) | web.archive.org s |
us-ny.gw.ivpn.net [109ms | OpenVPN] github.com s (ping) | web.archive.org f (ping) | idmsa.apple.com s (ping) | idmsa.apple.com s (ec: 8) | web.archive.org s |
us-az.gw.ivpn.net [167ms | OpenVPN] github.com s (ping) | web.archive.org f (ping) | idmsa.apple.com f (ping) | idmsa.apple.com s (ec: 8) | web.archive.org s |
nl.wg.ivpn.net [43ms | WireGuard] github.com s (ping) | web.archive.org f (ping) | idmsa.apple.com f (ping) | idmsa.apple.com s (ec: 4) | web.archive.org s |
lu.wg.ivpn.net [50ms | WireGuard] github.com s (ping) | web.archive.org f (ping) | idmsa.apple.com f (ping) | idmsa.apple.com s (ec: 4) | web.archive.org s |
ch.gw.ivpn.net [56ms | OpenVPN] github.com s (ping) | web.archive.org f (ping) | idmsa.apple.com f (ping) | idmsa.apple.com s (ec: 8) | web.archive.org s |
us-ca.wg.ivpn.net [186ms | WireGuard] github.com s (ping) | web.archive.org f (ping) | idmsa.apple.com f (ping) | idmsa.apple.com s (ec: 4) | web.archive.org s |
gb.gw.ivpn.net [55ms | OpenVPN] github.com s (ping) | web.archive.org f (ping) | idmsa.apple.com f (ping) | idmsa.apple.com s (ec: 8) | web.archive.org s |
rs.gw.ivpn.net [62ms | OpenVPN] github.com s (ping) | web.archive.org f (ping) | idmsa.apple.com s (ping) | idmsa.apple.com s (ec: 8) | web.archive.org s |
tw.gw.ivpn.net [254ms | OpenVPN] github.com s (ping) | web.archive.org f (ping) | idmsa.apple.com s (ping) | idmsa.apple.com s (ec: 8) | web.archive.org s |
hr.wg.ivpn.net [67ms | WireGuard] github.com s (ping) | web.archive.org f (ping) | idmsa.apple.com f (ping) | idmsa.apple.com s (ec: 4) | web.archive.org s |
us-il.gw.ivpn.net [187ms | OpenVPN] github.com s (ping) | web.archive.org f (ping) | idmsa.apple.com s (ping) | idmsa.apple.com s (ec: 8) | web.archive.org s |
ca-bc.wg.ivpn.net [166ms | WireGuard] github.com s (ping) | web.archive.org f (ping) | idmsa.apple.com f (ping) | idmsa.apple.com s (ec: 4) | web.archive.org s |
ua.wg.ivpn.net [144ms | WireGuard] github.com s (ping) | web.archive.org f (ping) | idmsa.apple.com f (ping) | idmsa.apple.com s (ec: 4) | web.archive.org s |
it.wg.ivpn.net [69ms | WireGuard] github.com s (ping) | web.archive.org f (ping) | idmsa.apple.com s (ping) | idmsa.apple.com s (ec: 4) | web.archive.org s |
no.wg.ivpn.net [29ms | WireGuard] github.com s (ping) | web.archive.org f (ping) | idmsa.apple.com f (ping) | idmsa.apple.com s (ec: 4) | web.archive.org s |
us-wa.gw.ivpn.net [180ms | OpenVPN] github.com s (ping) | web.archive.org f (ping) | idmsa.apple.com f (ping) | idmsa.apple.com s (ec: 8) | web.archive.org s |
sg.wg.ivpn.net [188ms | WireGuard] github.com s (ping) | web.archive.org f (ping) | idmsa.apple.com f (ping) | idmsa.apple.com s (ec: 4) | web.archive.org s |
za.wg.ivpn.net [227ms | WireGuard] github.com s (ping) | web.archive.org f (ping) | idmsa.apple.com s (ping) | idmsa.apple.com s (ec: 4) | web.archive.org s |
il.gw.ivpn.net [114ms | OpenVPN] github.com s (ping) | web.archive.org f (ping) | idmsa.apple.com f (ping) | idmsa.apple.com s (ec: 8) | web.archive.org s |
sk.gw.ivpn.net [68ms | OpenVPN] github.com f (ping) | web.archive.org f (ping) | idmsa.apple.com f (ping) | idmsa.apple.com s (ec: 4) | web.archive.org s (ec: 4) |
pl.wg.ivpn.net [76ms | WireGuard] github.com s (ping) | web.archive.org f (ping) | idmsa.apple.com s (ping) | idmsa.apple.com s (ec: 4) | web.archive.org s |
us-tx.wg.ivpn.net [153ms | WireGuard] github.com s (ping) | web.archive.org f (ping) | idmsa.apple.com s (ping) | idmsa.apple.com s (ec: 4) | web.archive.org s |
hk.wg.ivpn.net [194ms | WireGuard] github.com s (ping) | web.archive.org f (ping) | idmsa.apple.com s (ping) | idmsa.apple.com s (ec: 4) | web.archive.org s |
pt.gw.ivpn.net [84ms | OpenVPN] github.com s (ping) | web.archive.org f (ping) | idmsa.apple.com f (ping) | idmsa.apple.com s (ec: 8) | web.archive.org s |
pe.gw.ivpn.net [208ms | OpenVPN] github.com s (ping) | web.archive.org f (ping) | idmsa.apple.com f (ping) | idmsa.apple.com s (ec: 8) | web.archive.org s |
ua.gw.ivpn.net [153ms | OpenVPN] github.com s (ping) | web.archive.org f (ping) | idmsa.apple.com f (ping) | idmsa.apple.com s (ec: 4) | web.archive.org s |
au-nsw.wg.ivpn.net [284ms | WireGuard] github.com s (ping) | web.archive.org f (ping) | idmsa.apple.com f (ping) | idmsa.apple.com s (ec: 4) | web.archive.org s |
ca-qc.gw.ivpn.net [121ms | OpenVPN] github.com s (ping) | web.archive.org f (ping) | idmsa.apple.com s (ping) | idmsa.apple.com s (ec: 8) | web.archive.org s |
dk.gw.ivpn.net [39ms | OpenVPN] github.com s (ping) | web.archive.org f (ping) | idmsa.apple.com f (ping) | idmsa.apple.com s (ec: 8) | web.archive.org s |
us-va.gw.ivpn.net [132ms | OpenVPN] github.com s (ping) | web.archive.org f (ping) | idmsa.apple.com s (ping) | idmsa.apple.com s (ec: 8) | web.archive.org s |
sk.wg.ivpn.net [68ms | WireGuard] github.com s (ping) | web.archive.org f (ping) | idmsa.apple.com s (ping) | idmsa.apple.com s (ec: 4) | web.archive.org s |
us-fl.gw.ivpn.net [141ms | OpenVPN] github.com s (ping) | web.archive.org f (ping) | idmsa.apple.com s (ping) | idmsa.apple.com s (ec: 8) | web.archive.org s |
at.gw.ivpn.net [59ms | OpenVPN] github.com s (ping) | web.archive.org f (ping) | idmsa.apple.com s (ping) | idmsa.apple.com s (ec: 8) | web.archive.org s |
es.wg.ivpn.net [77ms | WireGuard] github.com s (ping) | web.archive.org f (ping) | idmsa.apple.com s (ping) | idmsa.apple.com s (ec: 4) | web.archive.org s |
bg.wg.ivpn.net [68ms | WireGuard] github.com s (ping) | web.archive.org f (ping) | idmsa.apple.com f (ping) | idmsa.apple.com s (ec: 4) | web.archive.org s |
us-az.wg.ivpn.net [166ms | WireGuard] github.com s (ping) | web.archive.org f (ping) | idmsa.apple.com f (ping) | idmsa.apple.com s (ec: 4) | web.archive.org s |
gr.wg.ivpn.net [96ms | WireGuard] github.com s (ping) | web.archive.org f (ping) | idmsa.apple.com s (ping) | idmsa.apple.com s (ec: 4) | web.archive.org s |
es.gw.ivpn.net [78ms | OpenVPN] github.com s (ping) | web.archive.org f (ping) | idmsa.apple.com s (ping) | idmsa.apple.com s (ec: 8) | web.archive.org s |
sg.gw.ivpn.net [187ms | OpenVPN] github.com s (ping) | web.archive.org f (ping) | idmsa.apple.com f (ping) | idmsa.apple.com s (ec: 8) | web.archive.org s |
gb-man.gw.ivpn.net [55ms | OpenVPN] github.com s (ping) | web.archive.org f (ping) | idmsa.apple.com f (ping) | idmsa.apple.com s (ec: 8) | web.archive.org s |
tw.wg.ivpn.net [241ms | WireGuard] github.com s (ping) | web.archive.org f (ping) | idmsa.apple.com s (ping) | idmsa.apple.com s (ec: 8) | web.archive.org s |
us-nv.wg.ivpn.net [167ms | WireGuard] github.com s (ping) | web.archive.org f (ping) | idmsa.apple.com s (ping) | idmsa.apple.com s (ec: 8) | web.archive.org s |
bg.gw.ivpn.net [68ms | OpenVPN] github.com s (ping) | web.archive.org f (ping) | idmsa.apple.com f (ping) | idmsa.apple.com s (ec: 8) | web.archive.org s |
dk.wg.ivpn.net [39ms | WireGuard] github.com s (ping) | web.archive.org f (ping) | idmsa.apple.com f (ping) | idmsa.apple.com s (ec: 4) | web.archive.org s |
fi.gw.ivpn.net [17ms | OpenVPN] github.com s (ping) | web.archive.org f (ping) | idmsa.apple.com f (ping) | idmsa.apple.com s (ec: 8) | web.archive.org s |
us-ut.wg.ivpn.net [198ms | WireGuard] github.com s (ping) | web.archive.org f (ping) | idmsa.apple.com s (ping) | idmsa.apple.com s (ec: 4) | web.archive.org s |
my.wg.ivpn.net [292ms | WireGuard] github.com s (ping) | web.archive.org f (ping) | idmsa.apple.com s (ping) | idmsa.apple.com s (ec: 4) | web.archive.org s |
be.wg.ivpn.net [45ms | WireGuard] github.com s (ping) | web.archive.org f (ping) | idmsa.apple.com s (ping) | idmsa.apple.com s (ec: 4) | web.archive.org s |
it.gw.ivpn.net [70ms | OpenVPN] github.com s (ping) | web.archive.org f (ping) | idmsa.apple.com s (ping) | idmsa.apple.com s (ec: 8) | web.archive.org s |
ca-qc.wg.ivpn.net [137ms | WireGuard] github.com s (ping) | web.archive.org f (ping) | idmsa.apple.com s (ping) | idmsa.apple.com s (ec: 4) | web.archive.org s |
nl.gw.ivpn.net [43ms | OpenVPN] github.com s (ping) | web.archive.org f (ping) | idmsa.apple.com s (ping) | idmsa.apple.com s (ec: 8) | web.archive.org s |
gb.wg.ivpn.net [55ms | WireGuard] github.com s (ping) | web.archive.org f (ping) | idmsa.apple.com f (ping) | idmsa.apple.com s (ec: 4) | web.archive.org s |
jp.wg.ivpn.net [278ms | WireGuard] github.com s (ping) | web.archive.org f (ping) | idmsa.apple.com f (ping) | idmsa.apple.com s (ec: 8) | web.archive.org s |
us-ma.gw.ivpn.net [115ms | OpenVPN] github.com s (ping) | web.archive.org f (ping) | idmsa.apple.com s (ping) | idmsa.apple.com s (ec: 8) | web.archive.org s |
us-va.wg.ivpn.net [139ms | WireGuard] github.com s (ping) | web.archive.org f (ping) | idmsa.apple.com s (ping) | idmsa.apple.com s (ec: 4) | web.archive.org s |
lu.gw.ivpn.net [48ms | OpenVPN] github.com s (ping) | web.archive.org f (ping) | idmsa.apple.com f (ping) | idmsa.apple.com s (ec: 8) | web.archive.org s |
fi.wg.ivpn.net [22ms | WireGuard] github.com s (ping) | web.archive.org f (ping) | idmsa.apple.com f (ping) | idmsa.apple.com s (ec: 4) | web.archive.org s |
il.wg.ivpn.net [108ms | WireGuard] github.com s (ping) | web.archive.org f (ping) | idmsa.apple.com f (ping) | idmsa.apple.com s (ec: 4) | web.archive.org s |
us-nj.gw.ivpn.net [110ms | OpenVPN] github.com s (ping) | web.archive.org f (ping) | idmsa.apple.com f (ping) | idmsa.apple.com s (ec: 8) | web.archive.org s |
us-ga.wg.ivpn.net [150ms | WireGuard] github.com s (ping) | web.archive.org f (ping) | idmsa.apple.com f (ping) | idmsa.apple.com s (ec: 4) | web.archive.org s |
se.wg.ivpn.net [44ms | WireGuard] github.com s (ping) | web.archive.org f (ping) | idmsa.apple.com s (ping) | idmsa.apple.com s (ec: 4) | web.archive.org s |
br.wg.ivpn.net [223ms | WireGuard] github.com s (ping) | web.archive.org f (ping) | idmsa.apple.com f (ping) | idmsa.apple.com s (ec: 4) | web.archive.org s |
ie.gw.ivpn.net [66ms | OpenVPN] github.com s (ping) | web.archive.org f (ping) | idmsa.apple.com s (ping) | idmsa.apple.com s (ec: 8) | web.archive.org s |
fr.gw.ivpn.net [63ms | OpenVPN] github.com s (ping) | web.archive.org f (ping) | idmsa.apple.com f (ping) | idmsa.apple.com s (ec: 8) | web.archive.org s |
cz.wg.ivpn.net [61ms | WireGuard] github.com s (ping) | web.archive.org f (ping) | idmsa.apple.com s (ping) | idmsa.apple.com s (ec: 4) | web.archive.org s |
is.wg.ivpn.net [78ms | WireGuard] github.com s (ping) | web.archive.org f (ping) | idmsa.apple.com s (ping) | idmsa.apple.com s (ec: 4) | web.archive.org s |
ie.wg.ivpn.net [65ms | WireGuard] github.com s (ping) | web.archive.org f (ping) | idmsa.apple.com s (ping) | idmsa.apple.com s (ec: 4) | web.archive.org s |
ca-on.gw.ivpn.net [124ms | OpenVPN] github.com s (ping) | web.archive.org f (ping) | idmsa.apple.com s (ping) | idmsa.apple.com s (ec: 8) | web.archive.org s |
us-nv.gw.ivpn.net [169ms | OpenVPN] github.com s (ping) | web.archive.org f (ping) | idmsa.apple.com s (ping) | idmsa.apple.com s (ec: 8) | web.archive.org s |
us-ny.wg.ivpn.net [109ms | WireGuard] github.com s (ping) | web.archive.org f (ping) | idmsa.apple.com s (ping) | idmsa.apple.com s (ec: 8) | web.archive.org s |
us-ga.gw.ivpn.net [138ms | OpenVPN] github.com s (ping) | web.archive.org f (ping) | idmsa.apple.com s (ping) | idmsa.apple.com s (ec: 8) | web.archive.org s |
jp.gw.ivpn.net [279ms | OpenVPN] github.com s (ping) | web.archive.org f (ping) | idmsa.apple.com f (ping) | idmsa.apple.com s (ec: 8) | web.archive.org s |
us-wa.wg.ivpn.net [180ms | WireGuard] github.com s (ping) | web.archive.org f (ping) | idmsa.apple.com f (ping) | idmsa.apple.com s (ec: 4) | web.archive.org s |
at.wg.ivpn.net [58ms | WireGuard] github.com s (ping) | web.archive.org f (ping) | idmsa.apple.com s (ping) | idmsa.apple.com s (ec: 4) | web.archive.org s |
rs.wg.ivpn.net [61ms | WireGuard] github.com s (ping) | web.archive.org f (ping) | idmsa.apple.com s (ping) | idmsa.apple.com s (ec: 4) | web.archive.org s |
gr.gw.ivpn.net [104ms | OpenVPN] github.com s (ping) | web.archive.org f (ping) | idmsa.apple.com f (ping) | idmsa.apple.com s (ec: 4) | web.archive.org s (ec: 4) |
mx.gw.ivpn.net [186ms | OpenVPN] github.com s (ping) | web.archive.org f (ping) | idmsa.apple.com f (ping) | idmsa.apple.com s (ec: 8) | web.archive.org s |
us-nj.wg.ivpn.net [111ms | WireGuard] github.com s (ping) | web.archive.org f (ping) | idmsa.apple.com f (ping) | idmsa.apple.com s (ec: 4) | web.archive.org s |
de.gw.ivpn.net [43ms | OpenVPN] github.com s (ping) | web.archive.org f (ping) | idmsa.apple.com s (ping) | idmsa.apple.com s (ec: 8) | web.archive.org s |
se.gw.ivpn.net [23ms | OpenVPN] github.com s (ping) | web.archive.org f (ping) | idmsa.apple.com s (ping) | idmsa.apple.com s (ec: 8) | web.archive.org s |
ro.wg.ivpn.net [67ms | WireGuard] github.com s (ping) | web.archive.org f (ping) | idmsa.apple.com f (ping) | idmsa.apple.com s (ec: 4) | web.archive.org s |
us-co.wg.ivpn.net [168ms | WireGuard] github.com s (ping) | web.archive.org f (ping) | idmsa.apple.com f (ping) | idmsa.apple.com s (ec: 4) | web.archive.org s |
my.gw.ivpn.net [293ms | OpenVPN] github.com s (ping) | web.archive.org f (ping) | idmsa.apple.com s (ping) | idmsa.apple.com s (ec: 8) | web.archive.org s |
za.gw.ivpn.net [232ms | OpenVPN] github.com s (ping) | web.archive.org f (ping) | idmsa.apple.com s (ping) | idmsa.apple.com s (ec: 8) | web.archive.org s |
no.gw.ivpn.net [29ms | OpenVPN] github.com s (ping) | web.archive.org f (ping) | idmsa.apple.com f (ping) | idmsa.apple.com s (ec: 8) | web.archive.org s |
us-fl.wg.ivpn.net [142ms | WireGuard] github.com s (ping) | web.archive.org f (ping) | idmsa.apple.com s (ping) | idmsa.apple.com s (ec: 4) | web.archive.org s |
us-ma.wg.ivpn.net [116ms | WireGuard] github.com s (ping) | web.archive.org f (ping) | idmsa.apple.com s (ping) | idmsa.apple.com s (ec: 4) | web.archive.org s |
be.gw.ivpn.net [45ms | OpenVPN] github.com s (ping) | web.archive.org f (ping) | idmsa.apple.com s (ping) | idmsa.apple.com s (ec: 8) | web.archive.org s |
us-ca.gw.ivpn.net [189ms | OpenVPN] github.com s (ping) | web.archive.org f (ping) | idmsa.apple.com f (ping) | idmsa.apple.com s (ec: 8) | web.archive.org s |
us-il.wg.ivpn.net [152ms | WireGuard] github.com s (ping) | web.archive.org f (ping) | idmsa.apple.com f (ping) | idmsa.apple.com s (ec: 8) | web.archive.org s |
ca-on.wg.ivpn.net [121ms | WireGuard] github.com s (ping) | web.archive.org f (ping) | idmsa.apple.com s (ping) | idmsa.apple.com s (ec: 8) | web.archive.org s |
fr.wg.ivpn.net [62ms | WireGuard] github.com s (ping) | web.archive.org f (ping) | idmsa.apple.com f (ping) | idmsa.apple.com s (ec: 4) | web.archive.org s |
cz.gw.ivpn.net [63ms | OpenVPN] github.com s (ping) | web.archive.org f (ping) | idmsa.apple.com s (ping) | idmsa.apple.com s (ec: 8) | web.archive.org s |
mx.wg.ivpn.net [191ms | WireGuard] github.com s (ping) | web.archive.org f (ping) | idmsa.apple.com f (ping) | idmsa.apple.com s (ec: 4) | web.archive.org s |
hu.wg.ivpn.net [89ms | WireGuard] github.com s (ping) | web.archive.org f (ping) | idmsa.apple.com s (ping) | idmsa.apple.com s (ec: 8) | web.archive.org s |
de.wg.ivpn.net [43ms | WireGuard] github.com s (ping) | web.archive.org f (ping) | idmsa.apple.com f (ping) | idmsa.apple.com s (ec: 8) | web.archive.org s |
```

</details>

```bash

#!/bin/bash

declare -A servers
declare -A server_ping
declare -A wget_exit_codes=(
    [0]="Success"
    [1]="Generic error"
    [2]="Parse error"
    [3]="File I/O error"
    [4]="Network failure"
    [5]="SSL verification failure"
    [6]="DNS lookup failure"
    [7]="Redirect loop detected"
    [8]="Server issued an error response"
    [9]="FTP-related error"
    [10]="No matching proxy found"
    [11]="Authentication failed"
    [20]="HTTP request writing error"
    [21]="Timeout reached"
    [22]="HTTP response error"
    [23]="Permission denied"
    [24]="Disk full"
    [25]="SSL certificate error or unsupported feature"
)

server_info_file="servers_info.txt";
output_file="ping_info.txt";
skip_servers=("" "");

start_server="";
start_processing=false;
if [[ -z "$start_server" ]]; then start_processing=true; fi;

ivpn disconnect

if [[ -f "$server_info_file" ]]; then
	echo "Reading servers info from file: $server_info_file";
	servers_info=$(cat "$server_info_file");
else
	echo "Fetching servers info and saving to file: $server_info_file"
	servers_info=$(ivpn servers -ping | awk -F'|' 'NR > 2 { gsub(/^[ \t]+|[ \t]+$/, "", $1); gsub(/^[ \t]+|[ \t]+$/, "", $2); gsub(/^[ \t]+|[ \t]+$/, "", $7); if (($1 == "OpenVPN" || $1 == "WireGuard") && $7 != "?") print $2, $1, $7 }');
	echo "$servers_info" > "$server_info_file";
fi;

while read -r server protocol ping; do servers["$server"]="$protocol"; server_ping["$server"]="$ping"; done <<< "$servers_info";

for server in "${!servers[@]}"; do
	if [[ "$start_processing" == false ]]; then if [[ "$server" == "$start_server" ]]; then echo "Starting processing from server: $server"; start_processing=true; else echo "Skipping server: $server"; continue; fi; fi;

	if [[ " ${skip_servers[*]} " == *" $server "* ]]; then echo "Skipping blacklisted server: $server"; continue; fi;

	protocol="${servers[$server]}";
    ping="${server_ping[$server]}"

	if [[ "$protocol" == "OpenVPN" ]]; then
		echo "Connecting to OpenVPN server: $server"
		timeout 15 ivpn connect -obfsproxy obfs4_iat_paranoid "$server"
	elif [[ "$protocol" == "WireGuard" ]]; then
		echo "Connecting to WireGuard server: $server"
		timeout 15 ivpn connect -v2ray quic -port "UDP:2049" "$server"
	fi

	if [ $? -ne 0 ]; then  echo "Connection to $server timed out or failed. Skipping."; continue; else echo "success"; fi;

	results="";
	for target in github.com web.archive.org idmsa.apple.com; do
		if ping -c 2 -W 1 "$target" > /dev/null 2>&1; then results+="$target s (ping) | "; else results+="$target f (ping) | "; fi;
	done;

	# results="${results%"${results##*[![:space:]]}"}";

	for target in idmsa.apple.com web.archive.org; do
		if wget --spider --timeout=2 --tries=1 "$target" > /dev/null 2>&1; then results+="$target s | ";
		else
			exit_code=$?; description=${wget_exit_codes[$exit_code]:-"Unknown error"};

			case $exit_code in
				8) results+="$target s (ec: $exit_code) | "; ;;
				4|5|6|7) results+="$target f (ec: $exit_code) | "; ;;
				*) results+="$target f (Exit Code: $exit_code, $description) | "; ;;
			esac;
		fi;
	done;


	results="${results%"${results##*[![:space:]]}"}";

	echo "$server [$ping | $protocol] $results" >> "$output_file";
done;

```

1) IVPN lacks a timeout parameter.
2) Cannot combine v2ray and obf parameters. Different server types (e.g., WireGuard doesn’t support obf, but v2ray is an option), obf should have higher priority than v2ray.
3) Port option for obf can become anything (even unlisted ranges). Manually specifying unlisted ranges fails despite them being possible in other cases.
4) Some servers don’t allow connection with obfuscation in paranoid mode, these servers allow obfuscation in less strict modes and even without obfuscation, some do not work with obfuscation at all. Doesn't matter if it’s v2ray or not.
5) Some servers aren't available to connect to on default ports so i had to specify a certain one for wireguard (can be fixed on your side?)

check the availability in the attached list, most of the vpn locations fail to create a ssl handshakes (`Unable to establish SSL connection`) with apple servers and many more services while others work fine, as you can see some servers aren't available for ping while they are available with wget, on different locations both ping and wget are available.

**Comment by ng8weji9vowdv** [comment]
also can you add an additional server for helsinki? it gets overloaded too quick, despite having low latency it barely can load pages lol (openvpn), same with norway (openvpn)

**Comment by ng8weji9vowdv** [comment]
ill update results later
